corgi/tasks/brew.py

```diff
@@ -332,6 +331,7 @@ def save_container(softwarebuild, build_data) -> ComponentNode:
         for module in build_data["meta"]["upstream_go_modules"]:
             new_upstream, created = Component.objects.get_or_create(
                 type=Component.Type.GOLANG,
+                namespace=Component.Namespace.UPSTREAM,
```

Comment:
The `upstream_go_modules` dict is actually set in the collector in:

```python
            # Legacy OSBS builds such as 1890187 copy source code into dist-git but specify where
            # the source code came from using the 'go' stanza in container.yaml
            # ref: https://osbs.readthedocs.io/en/osbs_ocp3/users.html#go
            go_modules = []
            if "go" in build_info["extra"]["image"]:
                for module in build_info["extra"]["image"]["go"]["modules"]:
                    if "module" in module:
                        go_modules.append(module["module"])
            if go_modules:
                component["meta"]["upstream_go_modules"] = go_modules
```

That's a different code path from the one that sets up proper Go components (which will have their `meta` object stored since it goes through the other path that checks `sources` (`...defaults={"meta_attr": source["meta"]...`). So for these modules we don't have much information other than the name. @jasinner, do we need to store anything more in the above `go_modules` list?

Reply:
Unfortunately those legacy OSBS builds are lacking in upstream metadata. The go stanza is really all we have to go on. There was actually a whole project [retrodep](https://github.com/release-engineering/retrodep) around trying to reverse engineer the commit in use. I seem to recall that project wrote the information into the container image. There builds are all pretty old though, so I don't think we need to extract that.